tests/unit_tests/repository_tests/test_pagination.py

```diff
@@ -121,3 +115,79 @@ def test_pagination_guides(database: AllRepositories, unique_user: TestUser):
+        SaveIngredientUnit(name="test unit 1", group_id=unique_user.group_id, use_abbreviation=True)
+    )
+
+    # wait a moment so we can test datetime filters
+    time.sleep(0.25)
+
+    unit_2 = database.ingredient_units.create(
+        SaveIngredientUnit(name="test unit 2", group_id=unique_user.group_id, use_abbreviation=False)
+    )
+
+    # wait a moment so we can test datetime filters
+    time.sleep(0.25)
+
+    unit_3 = database.ingredient_units.create(
+        SaveIngredientUnit(name="test unit 3", group_id=unique_user.group_id, use_abbreviation=False)
+    )
+
+    unit_ids = [unit.id for unit in [unit_1, unit_2, unit_3]]
+    units_repo = database.ingredient_units.by_group(unique_user.group_id)  # type: ignore
+
+    # make sure we can get all of our test units
+    query = PaginationQuery(page=1, per_page=-1)
+    all_units = units_repo.page_all(query).items
+    assert len(all_units) == 3
+
+    for unit in all_units:
+        assert unit.id in unit_ids
+
+    # check that basic filtering is working
+    query = PaginationQuery(page=1, per_page=-1, query_filter='name="test unit 2"')
+    unit_results = units_repo.page_all(query).items
+    assert len(unit_results) == 1
+    assert unit_results[0].id == unit_2.id
+
+    # check that datetimes are resolving properly
+    dt = unit_2.created_at.isoformat()
+    query = PaginationQuery(page=1, per_page=-1, query_filter=f'createdAt>="{dt}"')
+    unit_results = units_repo.page_all(query).items
+    assert len(unit_results) == 2
+    assert unit_1.id not in [unit.id for unit in unit_results]
+
+    # check that booleans are resolving properly
+    query = PaginationQuery(page=1, per_page=-1, query_filter="useAbbreviation=true")
+    unit_results = units_repo.page_all(query).items
+    assert len(unit_results) == 1
+    assert unit_results[0].id == unit_1.id
+
+    # check an advanced filter
+    dt = unit_3.created_at.isoformat()
+    qf = f'name="test unit 1" OR (useAbbreviation=f AND (name="test unit 2" OR createdAt > "{dt}"))'
+    query = PaginationQuery(page=1, per_page=-1, query_filter=qf)
+    unit_results = units_repo.page_all(query).items
+    assert len(unit_results) == 2
+    assert unit_3.id not in [unit.id for unit in unit_results]
+
+    # verify that improper queries throw 400 errors
+    route = "/api/units"
+
+    # unbalanced parenthesis
+    qf = '(name="test name" AND useAbbreviation=f))'
+    response = api_client.get(route, params={"queryFilter": qf}, headers=unique_user.token)
+    assert response.status_code == 400
+
+    # invalid datetime format
+    qf = 'createdAt="this is not a valid datetime format"'
+    response = api_client.get(route, params={"queryFilter": qf}, headers=unique_user.token)
+    assert response.status_code == 400
+
+    # invalid attribute
+    qf = 'badAttribute="test value"'
+    response = api_client.get(route, params={"queryFilter": qf}, headers=unique_user.token)
+    assert response.status_code == 400
```

Comment:
> I'm happy to add some more tests! Pytest is a bit of an enigma to me but using what you've already done as reference has made it pretty easy. What sort of isolated cases did you have in mind? Just more specific stuff?

I think you cover most of the cases you've introduced. Basically I'm looking for each one of these comments to be their own test so it's easier to isolate specific cases that I end up breaking. You could probably accomplish this by running the setup in a fixture and injecting what you need into the function from there and just set the fixture up to only run once.